### DISCUSSION TITLE
Access event/application parameters from an observer?

### DISCUSSION THREAD

**Discussion by moPellowe** [Open-Systems-Pharmacology/Forum#1618] [Questions & Problems] [answered]
Is it possible to access application parameters (for example parameters from the ProtocolSchemaItem container) within an Observer? I noticed that Applications does not show up under the "Possible referenced objects" window pane, so I have written the path manually, but the observer is not showing up within the simulation.

**Comment by PavelBal** [comment]
Could you paste here the path that you are using?

**Reply by moPellowe** [reply]
After trying to recreate the issue, I see that I can actually access "Application" parameters. The parameter that seems to be causing the issue is "Time" which I had set to reference TIME. Is this permitted within an Observer?

**Reply by PavelBal** [reply]
Hmm no, I do not think so.. But it is hard to tell without seeing what exactly you are trying to do :)

**Reply by moPellowe** [reply]
I have tried a minimal example where I am trying to access TIME within an observer. Please find the MoBi file attached.
![image](https://github.com/Open-Systems-Pharmacology/Forum/assets/57540473/91d7168b-dc64-4926-ba08-bb9f65ccc6b6)
[test.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/12611307/test.zip)

**Reply by PavelBal** [reply]
Seems that this is not possible, indeed... @msevestre @Yuri05 Is there a reason for it (short version - `TIME` cannot be used in observer formulas)

**Reply by Yuri05** [reply]
No reason - must be a bug.
As a workaround: create a parameter in the spatial structure (e.g. under Organism) and set its formula to `Time` - then use the parameter in observers.

**Reply by moPellowe** [reply]
I can see how that workaround would work, but I'm wanting to bring the Observer back to PK-Sim, so I ideally need a solution that is self-contained within the Observer. What I am trying to do is calculate the amount of drug that has been administered from t=0 to t=TIME for an infusion.

**Comment by Yuri05** [comment, marked as answer]
> I can see how that workaround would work, but I'm wanting to bring the Observer back to PK-Sim, so I ideally need a solution that is self-contained within the Observer. What I am trying to do is calculate the amount of drug that has been administered from t=0 to t=TIME for an infusion.

You could define such an observer as e.g. "**Initial drug mass** - **remaining drug mass**" 
[Administered.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/12618588/Administered.zip)

![grafik](https://github.com/Open-Systems-Pharmacology/Forum/assets/25061876/293b83be-0bfe-495e-b8a7-4dbbfd55e7ff)

![grafik](https://github.com/Open-Systems-Pharmacology/Forum/assets/25061876/2a82a685-ee8e-47d0-a352-23b0a19c5223)

<img width="250" src="https://github.com/Open-Systems-Pharmacology/Forum/assets/25061876/5ae92b8d-3056-4023-8d57-0aea043b5f73"> <img width="600" src="https://github.com/Open-Systems-Pharmacology/Forum/assets/25061876/303bfe27-40dc-4ccf-901b-bf1ae4069ad7">

**Reply by moPellowe** [reply]
Thank you, @Yuri05 , for this solution. This is what I needed.
